File: src/test/java/DoubleAlternateTests.java
import com.github.pauldambra.moduluschecker.Account.BankAccount;
import com.github.pauldambra.moduluschecker.ModulusAlgorithm;
import com.github.pauldambra.moduluschecker.ModulusCheckParams;
import com.github.pauldambra.moduluschecker.chain.checks.DoubleAlternateCheck;
import com.github.pauldambra.moduluschecker.valacdosFile.WeightRow;
import com.google.common.collect.ImmutableList;
import org.junit.Test;

import java.util.Optional;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.core.IsEqual.equalTo;

public class DoubleAlternateTests {

    @Test
    public void CanRunDoubleAlternateCheck() {
        String sc = "499273";
        String an = "12345678";
        WeightRow row = new WeightRow(
                ModulusAlgorithm.DOUBLE_ALTERNATE,
                ImmutableList.of(2,1,2,1,2,1,2,1,2,1,2,1,2,1),
                Optional.empty()
                );

        ModulusCheckParams params = new ModulusCheckParams(
                BankAccount.Of(sc, an),
                Optional.of(row),
                Optional.empty(),
                Optional.empty());
        DoubleAlternateCheck checker = new DoubleAlternateCheck();

        Boolean result = checker.check(params, x -> x.firstWeightRow.get());

        assertThat(result, is(equalTo(true)));
    }
}

File: src/test/java/DoubleAlternateTests.kt


import com.github.pauldambra.moduluschecker.ModulusAlgorithm
import com.github.pauldambra.moduluschecker.ModulusCheckParams
import com.github.pauldambra.moduluschecker.account.BankAccount
import com.github.pauldambra.moduluschecker.chain.checks.DoubleAlternateCheck
import com.github.pauldambra.moduluschecker.valacdosFile.WeightRow
import com.natpryce.hamkrest.assertion.assert
import com.natpryce.hamkrest.equalTo
import org.junit.Test

class DoubleAlternateTests {

    @Test
    fun canRunDoubleAlternateCheck() {
        val sc = "499273"
        val an = "12345678"
        val row = WeightRow(
          ModulusAlgorithm.DOUBLE_ALTERNATE,
          listOf(2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1),
          null
        )

        val params = ModulusCheckParams(BankAccount(sc, an), row)
        val checker = DoubleAlternateCheck()

        val result = checker.check(params.account, params.firstWeightRow!!)

        assert.that(result, equalTo(true))
    }
}
